core-blocks/playlist/edit.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Button,
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	CheckboxControl,
+	PanelBody,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	InspectorControls,
+	MediaUpload,
+	BlockControls,
+	editorMediaUpload,
+} from '@wordpress/editor';
+
+/**
+ * Internal dependencies
+ */
+import './editor.scss';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+
+		// check for if ids is set to determine edit state
+		this.state = {
+			editing: ! this.props.attributes.ids,
+		};
+	}
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
+	}
+
+	render() {
+		const { attributes, setAttributes, className } = this.props;
+		const { editing } = this.state;
+		const { tracklist, artists, images } = attributes;
+
+		const onSelectMedia = ( media ) => {
+			//check if there are returned media items and set attributes when there are
+			if ( media && media[ 0 ].url ) {
+				media = ( 1 < media.length ) ? media : [ media ];
+				const ids = JSON.stringify( media.map( ( item ) => item.id ) );
+				setAttributes( { tracklist, artists, images, ids, type: media[ 0 ].type } );
+				this.setState( { editing: false } );
+			}
+		};
+
+		const setAudio = ( [ audio ] ) => onSelectMedia( audio );
+
+		const uploadFromFiles = ( event ) => editorMediaUpload( event.target.files, setAudio, 'audio' );
```

Comment:
The only parameter of `editorMediaUpload` should be an object, I think this can solve the upload problem.

Reply:
maybe a simple fix....considering the existing behavior of the editor is to `add media` and then click playlist>upload, do we really need an upload button? On initial click of the media button, you have the option to upload in the first tab @karmatosed would you agree? The upload button was copied from a early iteration of audio

Reply:
btw, still working to get this fixed so we can have the option. Just want to verify that it is actually a wanted feature of the block